### PR TITLE
Response to reviews

### PR DESCRIPTION
**Pull request by ethanwhite**

### CODE REVIEW

**Comment by ethanwhite** [comment]
This is my in-progress revision and response letter. It's not a full draft yet, but it's where I got to before it was time to pickup River and take her to get her hair cut. Comments welcome or feel free to wait until a full draft is in.

**Comment by sdtaylor** [comment]
This reference can be used for the first revewers comment "Could this section include some examples that demonstrate the value of iterative near-term forecasting?" 

Specifically:
>NWP has a fundamental advantage over many other scientific disciplines in that its skill is objectively evaluated daily and globally, so that success and failure of forecasts is accurately known and pathways to improve predictive skill can be effectively tested.

```
@article{bauer2015,
  title={The quiet revolution of numerical weather prediction},
  author={Bauer, Peter and Thorpe, Alan and Brunet, Gilbert},
  journal={Nature},
  volume={525},
  number={7567},
  pages={47},
  year={2015},
  publisher={Nature Publishing Group}
}
```

**Comment by ethanwhite** [comment]
The revised version and response to reviewers is now a full draft. Comments welcome. We're happy to implement or feel free to PR into this branch.

**Comment by sdtaylor** [comment]
Looks great to me: :+1:

File: manuscript/response_to_review.Rmd [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+We disagree with the assumption that researchers that are proficient at modeling and coding would already have similar protocols and pipelines in place in their routine research?)". We are active in this field and aware of very few efforts of this scope in ecology. These researchers will have the foundational tools to develop similar systems (or contribute to this one).
```
Comment (juniperlsimonis) [inline manuscript/response_to_review.Rmd]: remove the " at the end of the sentence

File: manuscript/response_to_review.Rmd [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+There are a number of interesting points in this paragraph that we feel are beyond the scope of this current paper. We have already described the importance of new models and how we have facilitated adding them. We have now further expanded this section to describe how those external to the group can contribute improved models. Introducing and discussing the different implications of differnt forms of error and how the are associated with increasing in data vs. improvements in models is an interesting subject for forecasting, but is out of scope of this paper. It would definitely be something interesting to explore in future work.
```
Comment (juniperlsimonis) [inline manuscript/response_to_review.Rmd]: differnt -> different

File: manuscript/response_to_review.Rmd [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+**Minor comments:**
+   
+**Lines 63 to 83:  Could this section include some examples that demonstrate the value of iterative near-term forecasting from other fields? Could it be argued that iterative near term forecasting has led to progress in other fields? Say meteorology and political science?**
+
+We have now mentioned the established benefits for iterative near term forecasting in meteorology and added several citations for readers interested in exploring this further. 
+
+**Figure 1: Another point on figure one.  What about model improvement? It seems like there should be a feedback loop somewhere showing how models get refined, added, dropped etc.**
+
+We tried to add a visual representation of this to Figure 1, but could not make it work without making the figure overly crowded. We have added some additional discussion of this "Modeling and Forecasting" section of the paper.
```
Comment (juniperlsimonis) [inline manuscript/response_to_review.Rmd]: discussion of this "Modeling and Forecasting" -> discussion of this to the "Modeling and Forecasting"

File: manuscript/portal_iterative_forecasting_ms.Rmd [hunk 4]
```diff
@@ -41,13 +41,16 @@ Key-words: forecasting, prediction, mammals, iterative forecasting, Portal Proje
...
... state of ecological systems is important for management, conservation, and evaluation of how well models capture the processes governing ecological systems [@clark2001; @tallis2006; @diaz2015; @dietze2017]. In 2001, Clark et al. [-@clark2001] called for a more central role of forecasting in ecology. Since then, an increasing number of ecological forecasts are being published that focus on a societally important questions from daily to decadal time scales [@dietze2018]. At daily scales, ecological forecasts predict the occurance of environmental issues like toxic algal blooms [@stumpf2009]. At quarterly scales forecasts are used to predict the stocks of fisheries and make decisions about quotas for fishing [@NOAA2016]. At decadal time scales, ecological forecasts are used to predict how biodiversity will change as it responds to anthropogenic influences [@harris2018]. These forecasting examples highlight the important role that ecological forecasts play in recasting ecological knowledge in societally relevant ways and also improve our understanding of ecological systems by testing the ability of our models to predict how systems will change in the future [@dietze2018, @harris2018].
```
Comment (juniperlsimonis) [inline manuscript/portal_iterative_forecasting_ms.Rmd]: [@dietze2018, @harris2018] -> [@dietze2018; @harris2018]

File: manuscript/portal_iterative_forecasting_ms.Rmd [hunk 5]
```diff
@@ -41,13 +41,16 @@ Key-words: forecasting, prediction, mammals, iterative forecasting, Portal Proje
...
... updated [@dietze2018]. This lack of both regular assessment and active updating has limited the progress of ecological forecasting and hindered our ability to make useful and reliable predictions. The lack of active assessment results in limited information on how much confidence to place in forecasts and makes it difficult to determine on which forecasting methods to build. Without regular updates, forecasts lack the most current data, and the longer a forecast remains out of date, the less accurate it becomes [@petchey2015; @dietze2018]. More regular updating and assessment will advance ecological forecasting as a field by accelerating the identification of the best models for individual forecasts and improving our understanding of how to best design forecasting approaches for ecology in general. This approach has helped accelerate forecasting ability in other fields including meteorology [@kalnay2003, @mcgill2012, @bauer2015]. For ecological forecasting to mature as a field, we need to change how we produce and interact with forecasts, creating a more dynamic interplay between model development, prediction generation, and incorporation of new data and information [@dietze2018].
```
Comment (juniperlsimonis) [inline manuscript/portal_iterative_forecasting_ms.Rmd]: [@kalnay2003, @mcgill2012, @bauer2015] -> [@kalnay2003; @mcgill2012; @bauer2015]

File: manuscript/portal_iterative_forecasting_ms.Rmd [hunk 6]
```diff
@@ -41,13 +41,16 @@ Key-words: forecasting, prediction, mammals, iterative forecasting, Portal Proje
...
...14; @yenni2018]. We explored what it would entail to operationalize Dietze et al's recommendations by constructing our own iterative near-term forecasting pipeline for an on-going, long-term ecological study that collects high-frequency data on desert rodent abundances [@brown1998; @ernest2008]. We constructed an automated forecasting pipeline with the goal of being able to forecast rodent abundances and evaluate our predictions on a monthly basis. In this paper, we discuss our approach for creating this iterative near-term forecasting pipeline, the challenges we encountered, the tools we used, and the lessons we learned so that others can create their own iterative forecasting systems. For those interested in implementing iterative forecasting, either on their own or as part of a team, this paper will provide a roadmap for how to build such a system and what skills will be helpful to do so. For readers looking for an introduction to automation and continous integration in an ecological context, we recommend our paper on data management for continuously collected data, which includes a tutorial on how to set up some of the aspects of automation described in this paper (@yenni2018).
```
Comment (juniperlsimonis) [inline manuscript/portal_iterative_forecasting_ms.Rmd]: (@yenni2018) -> [@yenni2018]

File: manuscript/portal_iterative_forecasting_ms.Rmd [hunk 7]
```diff
@@ -57,11 +60,11 @@ Iterative forecasting is most effective with frequently collected data, since it
...
+![a) Stages of the forecasting pipeline. To go from raw data to forecast presentation involves a number of stages, each of which requires unique tasks, tools and infrastructure. The stages are interdependent, with outputs from one stage forming the inputs for the subsequent stage. Tasks in all stages are run using code written in R. b) Continuous integration system. Each box denotes the core infrastructure used for each stage of the forecasting pipeline. Continuous integration (denoted by the Travis icon, a woman wearing safety glasses and hardhat) triggers the code involved in events that link the stages of the pipeline, such as using the output from the forecasting stage (purple box) to create an updated website (rose box). Travis also runs tasks within a stage, such as testing code and adding weather data (icons on arrows originating and ending on the same box). The code for driving different stages of this pipeline is stored on GitHub (denoted by the GitHub icon, an "octocat").](figs/Figure1.png)
```
Comment (juniperlsimonis) [inline manuscript/portal_iterative_forecasting_ms.Rmd]: figure uses A B, caption uses a b

Comment (juniperlsimonis) [reply manuscript/portal_iterative_forecasting_ms.Rmd]: text uses a b too, so suggest changing it in the figure

File: manuscript/portal_iterative_forecasting_ms.Rmd [hunk 8]
```diff
@@ -57,11 +60,11 @@ Iterative forecasting is most effective with frequently collected data, since it
...
...es in the exact state used to develop and run the software. If those dependencies change (or disappear) in the wider world, they still exist, unchanged, in the container. We use an existing platform, Docker [@merkel2014], to store an exact image of the complete software environment for running the forecasts. Docker also allows a specified set of packages to be used consistently across different computer and server environments. Using containers allows us to control transitions to new package versions, implementing them only after we have tested them and made any necessary changes to the data processing and analysis code. We use a container created by the Rocker project, which is a Docker image with many important R packages (i.e. tidyverse) pre-installed  [@boettiger2017]. We add our code and dependencies to this existing Rocker image to create a software container for our forecasting pipeline. In combination, the automated running of the pipeline (continuous integration) and the guarantee it will not stop working unexpectedly due to software dependencies (via a software container) allows continuous analysis to serve as the glue that connects all stages of the forecasting pipeline.
```
Comment (juniperlsimonis) [inline manuscript/portal_iterative_forecasting_ms.Rmd]: `tscount` is used by *two* of our models

Comment (juniperlsimonis) [inline manuscript/portal_iterative_forecasting_ms.Rmd]: format and cite "tidyverse" like other packages

File: manuscript/portal_iterative_forecasting_ms.Rmd [hunk 9]
```diff
@@ -57,11 +60,11 @@ Iterative forecasting is most effective with frequently collected data, since it
...
```
Comment (juniperlsimonis) [inline manuscript/portal_iterative_forecasting_ms.Rmd]: "once data is" -> "once data are"

File: manuscript/portal_iterative_forecasting_ms.Rmd [hunk 10]
```diff
@@ -83,13 +86,13 @@ Once data is available, it must be processed into a form appropriate for modelin
...
...ently evaluated using root mean square error (RMSE) to evaluate point forecasts and coverage to evaluate uncertainty. We plan to add additional metrics in the future. In addition to evaluating the actual forecasts, we also use hindcasting [forecasting on already collected data; @jolliffe2003] to gain additional insight into the methods that work best for forecasting this system. For example, a model is fit using rodent observations up to June 2005, then used to make a forecast 12 months out to May 2006. The observations of that 12-month period can immediately be used to evaluate the model. Since hindcasting is conducted using data that has already been collected, it allows model comparisons to be conducted on large numbers of hindcasts and provides insight into which models make the best forecasts without needing to wait for new data to be collected [@harris2018]. It can also be used to quickly evaluate new models instead of waiting for an adequate amount of data to accumulate. As the performance of different models is understood through evaluation of forecasts and hindcasts models can be refined or removed from the system or ensemble to iteratively improve the resulting forecasts.
```
Comment (juniperlsimonis) [inline manuscript/portal_iterative_forecasting_ms.Rmd]: add comma between  "models" and "can" in last sentence

**Comment by ethanwhite** [comment]
Thanks @DAPPERstats! All your fixes are in.

**Comment by gmyenni** [comment]
Looks awesome! 👍

**Comment by emchristensen** [comment]
Looks great!

**Pull request merged**